### PR TITLE
Ingredient store

### PR DESCRIPTION
**Pull request by mam249**
Use custom search api and the inputted ingredients array to retrieve links from bbcgoodfood with the relevant recipe outputs

### CODE REVIEW

File: backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java
Comment (azhara-a) [inline backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java:37]: Update this comment to describe what the post request does

File: backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java
Comment (azhara-a) [inline backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java:48]: Make the link into a constant to improve readability or create a function getCustomSearchLink() that returns the link

File: frontend/public/index.html
Comment (azhara-a) [inline frontend/public/index.html:53]: Is that still needed?

**Review by azhara-a — COMMENTED** [review]
LGTM, left some nits. It might also be good to write a description in your pull request as it might be unclear for future reviewers.

File: backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java
Comment (azhara-a) [inline backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java:38]: Could you please make FindRecipesServlet extend AuthenticationServlet? This is to ensure unauthorised users don't end any requests. https://github.com/googleinterns/step-2020-recipe-finder/blob/master/backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java

File: backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java
Comment (azhara-a) [inline backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java:43]: FindRecipes servlet has to override post() and get() which are AuthenticationServlet's abstract methods. Please look at how AuthenticationServlet is implemented

Comment (azhara-a) [inline backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java:43]: I also think that this should be a post request rather than a get request. This is what we're doing on the frontend

File: backend/src/main/java/com/google/sps/scraping/BBCGoodFoodRecipeScraper.java
Comment (lminchillo) [inline backend/src/main/java/com/google/sps/scraping/BBCGoodFoodRecipeScraper.java:60]: - Add newline after line 60
- I think the comment above the function is incomplete, I didn't quite get what the method does just by reading the comment. Can you please re-write?

File: backend/src/main/java/com/google/sps/scraping/BBCGoodFoodRecipeScraper.java
Comment (lminchillo) [inline backend/src/main/java/com/google/sps/scraping/BBCGoodFoodRecipeScraper.java:109]:
```suggestion
    JsonElement dietElements = jsonObject.get("suitableForDiet");
    if (dietElements == null) {
      return new String[0];
    }
```


Instead of creating a variable `NO_DIET` (wrong naming convention for local variables) you can just return an empty array

File: backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java
Comment (lminchillo) [inline backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java:44]: remove extra indentation

File: frontend/src/components/InputText.js
Comment (lminchillo) [inline frontend/src/components/InputText.js:20]: nit: remove extra newline

File: frontend/src/components/Inventory.js
Comment (lminchillo) [inline frontend/src/components/Inventory.js:1]: can you please add the default google license to this (and all other files you created that don't have it?)

File: backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java
Comment (azhara-a) [inline backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java:49]: I thought this is how we retrieved ingredients: `String ingredients = request.getReader().readLine();`

File: frontend/src/App.js
Comment (azhara-a) [inline frontend/src/App.js]: Inventory and Shopping List are missing, I think it's due to the merge

File: frontend/src/components/InputText.js
Comment (azhara-a) [inline frontend/src/components/InputText.js]: Could you make the button disabled if user didn't input anything? It could be something like: `<Button disabled={this.state.items.length === 0} ..`

File: frontend/src/components/InputText.js
Comment (azhara-a) [inline frontend/src/components/InputText.js]: Make sure to add this. I think the merge removed it

**Review by azhara-a — CHANGES_REQUESTED** [review]
Merge with master removed a few lines

File: backend/src/main/java/com/google/sps/scraping/BBCGoodFoodRecipeScraper.java
Comment (azhara-a) [inline backend/src/main/java/com/google/sps/scraping/BBCGoodFoodRecipeScraper.java:56]: The comment should describe what the function does, so `Outputs link to the Custom Search API` or something similar is sufficient. No need to say `created this function`

File: backend/src/main/java/com/google/sps/scraping/BBCGoodFoodRecipeScraper.java
Comment (azhara-a) [inline backend/src/main/java/com/google/sps/scraping/BBCGoodFoodRecipeScraper.java:108]: Typo: `String[0]`

File: backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java
Comment (azhara-a) [inline backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java:36]: Update this comment to `in Post request, returns a list of recommended recipes based on the ingredients in the request`

File: backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java
Comment (azhara-a) [inline backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java:44]: Add space between // and no, so `// no get request`

File: backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java
Comment (azhara-a) [inline backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java:51]: Put the bracket `)` in the line above

File: backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java
Comment (azhara-a) [inline backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java:69]: Add `response.setCharacterEncoding("UTF8");` before setting content type. This will resolve the black diamonds issue

File: frontend/src/components/Inventory.js
Comment (azhara-a) [inline frontend/src/components/Inventory.js:17]: Remove unnecessary imports

File: frontend/src/components/individual-recipe/Tutorial.js
Comment (azhara-a) [inline frontend/src/components/individual-recipe/Tutorial.js:124]:
```suggestion
  finishCooking(recipe) {
```

File: frontend/src/components/individual-recipe/Tutorial.js
Comment (azhara-a) [inline frontend/src/components/individual-recipe/Tutorial.js:133]:
```suggestion
    const request = new Request("/api/store-recipe", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        Accept: "application/json",
      },
      body: JSON.stringify(recipe)
    });
```

File: frontend/src/components/individual-recipe/Tutorial.js
Comment (azhara-a) [reply frontend/src/components/individual-recipe/Tutorial.js:124]: It's because once you click on the button that redirects to another page, `this` doesn't refer to Tutorial.js, it will refer to something undefined. That's why you need to accept the recipe as a parameter.

**Review by azhara-a — APPROVED** [review]
LGTM

File: backend/src/main/java/com/google/sps/scraping/BBCGoodFoodRecipeScraper.java
Comment (lminchillo) [inline backend/src/main/java/com/google/sps/scraping/BBCGoodFoodRecipeScraper.java:108]: I don't think this works? It's missing `new`:
`return new String[0];`

File: frontend/src/components/ShoppingList.js
Comment (lminchillo) [inline frontend/src/components/ShoppingList.js:1]: Please add license

**Review by lminchillo — APPROVED** [review]
Good job! LGTM after fixing these 2 comments, I'll mark Approved so you can merge it after this

**Pull request merged**
